src/libraries/navdrawer/navdrawer.js

```diff
@@ -3,358 +3,373 @@
-    };
+    hideMask() {
+        this.mask.classList.add('hide');
+        this.mask.classList.remove('backdrop');
+    }
 
-    TouchMenuLA.prototype.invoke = function (fn) {
+    invoke(fn) {
         if (fn) {
-            fn.apply(self);
+            fn.apply(this);
         }
-    };
+    }
 
-    let _edgeSwipeEnabled;
+    setEdgeSwipeEnabled(enabled) {
+        const options = this.options;
 
-    TouchMenuLA.prototype.setEdgeSwipeEnabled = function (enabled) {
         if (!options.disableEdgeSwipe) {
             if (browser.touch) {
                 if (enabled) {
-                    if (!_edgeSwipeEnabled) {
-                        _edgeSwipeEnabled = true;
-                        dom.addEventListener(edgeContainer, 'touchstart', onEdgeTouchStart, {
+                    if (!this._edgeSwipeEnabled) {
+                        this._edgeSwipeEnabled = true;
+                        dom.addEventListener(this.edgeContainer, 'touchstart', this.onEdgeTouchStart, {
                             passive: true
                         });
-                        dom.addEventListener(edgeContainer, 'touchend', onEdgeTouchEnd, {
+                        dom.addEventListener(this.edgeContainer, 'touchend', this.onEdgeTouchEnd, {
                             passive: true
                         });
-                        dom.addEventListener(edgeContainer, 'touchcancel', onEdgeTouchEnd, {
+                        dom.addEventListener(this.edgeContainer, 'touchcancel', this.onEdgeTouchEnd, {
                             passive: true
                         });
                     }
                 } else {
-                    if (_edgeSwipeEnabled) {
-                        _edgeSwipeEnabled = false;
-                        dom.removeEventListener(edgeContainer, 'touchstart', onEdgeTouchStart, {
+                    if (this._edgeSwipeEnabled) {
+                        this._edgeSwipeEnabled = false;
+                        dom.removeEventListener(this.edgeContainer, 'touchstart', this.onEdgeTouchStart, {
                             passive: true
                         });
-                        dom.removeEventListener(edgeContainer, 'touchend', onEdgeTouchEnd, {
+                        dom.removeEventListener(this.edgeContainer, 'touchend', this.onEdgeTouchEnd, {
                             passive: true
                         });
-                        dom.removeEventListener(edgeContainer, 'touchcancel', onEdgeTouchEnd, {
+                        dom.removeEventListener(this.edgeContainer, 'touchcancel', this.onEdgeTouchEnd, {
                             passive: true
                         });
                     }
                 }
             }
         }
-    };
+    }
 
-    TouchMenuLA.prototype.initialize = function () {
-        options = Object.assign(defaults, options || {});
+    initialize() {
+        const options = Object.assign(this.defaults, this.options || {});
```

Comment:
We need to store the new options. Otherwise `handleSize` is `undefined`.
```suggestion
        const options = Object.assign({}, this.defaults, this.options || {});
        this.options = options;
```
Assignment to an empty object, so as not to change `defaults`. The latter can also be extracted as const.